search/searcher/search_contains_test.go

```diff
@@ -0,0 +1,1040 @@
+//  Copyright (c) 2022 Couchbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package searcher
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/blevesearch/bleve/v2/document"
+)
+
+var (
+	leftRectEdgeMultiPoint [][]float64   = [][]float64{{-1, 0.2}, {-0.9, 0.1}}
+	leftRectWithHole       [][][]float64 = [][][]float64{{{-1, 0}, {0, 0}, {0, 1}, {-1, 1}, {-1, 0}},
+		{{-0.75, 0.25}, {-0.75, -0.75}, {-0.25, 0.75}, {-0.25, 0.25}, {-0.74, 0.25}}}
+	leftRectEdgePoint  []float64   = []float64{-1, 0.2}
+	leftRectMultiPoint [][]float64 = [][]float64{{0.5, 0.5}, {-0.9, 0.1}}
+)
+
+func TestPointPolygonContains(t *testing.T) {
+	tests := []struct {
+		QueryShape       []float64
+		DocShapeVertices [][][]float64
+		DocShapeName     string
+		Expected         []string
+		Desc             string
+		QueryType        string
+	}{
+		{
+			QueryShape:       rightRectPoint,
+			DocShapeVertices: rightRect,
+			DocShapeName:     "polygon1",
+			Expected:         []string{"polygon1"},
+			Desc:             "point inside polygon",
+			QueryType:        "contains",
+		},
+		{
+			QueryShape:       leftRectPoint,
+			DocShapeVertices: nil,
+			DocShapeName:     "",
+			Expected:         nil,
+			Desc:             "empty polygon",
+			QueryType:        "contains",
+		},
+	}
+	i := setupIndex(t)
+
+	for _, test := range tests {
+		doc := document.NewDocument(test.DocShapeName)
+		doc.AddField(document.NewGeoShapeFieldWithIndexingOptions("geometry", []uint64{},
+			[][][][]float64{test.DocShapeVertices}, "polygon", document.DefaultGeoShapeIndexingOptions))
+		err := i.Update(doc)
```

Comment:
line 64-92 looks repeated across many tests and similar patterns are visible across tests.
Can this be another function with the right arguments and reduce the boilerplate code?

Reply:
In a recent WIP commit, I have attempted to do something like this for 1 func. I'm not _entirely_ sure about it. Could you PTAL and let me know?
thanks!